src/seq_mv/vector_device.c

```diff
@@ -151,6 +151,43 @@ hypre_SeqVectorAxpyDevice( HYPRE_Complex alpha,
    return hypre_error_flag;
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_SeqVectorAxpyzDevice
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_SeqVectorAxpyzDevice( HYPRE_Complex  alpha,
+                            hypre_Vector  *x,
+                            HYPRE_Complex  beta,
+                            hypre_Vector  *y,
+                            hypre_Vector  *z )
+{
+   HYPRE_Complex  *x_data      = hypre_VectorData(x);
+   HYPRE_Complex  *y_data      = hypre_VectorData(y);
+   HYPRE_Complex  *z_data      = hypre_VectorData(z);
+
+   HYPRE_Int       num_vectors = hypre_VectorNumVectors(x);
+   HYPRE_Int       size        = hypre_VectorSize(x);
+   HYPRE_Int       total_size  = size * num_vectors;
+
+#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP) || defined(HYPRE_USING_SYCL)
+   hypreDevice_ComplexAxpyzn(total_size, x_data, y_data, z_data, alpha, beta);
+
+#elif defined(HYPRE_USING_DEVICE_OPENMP)
+   HYPRE_Int i;
+
+   #pragma omp target teams distribute parallel for private(i) is_device_ptr(z_data, y_data, x_data)
+   for (i = 0; i < total_size; i++)
+   {
+      z_data[i] = alpha * x_data[i] + beta * z_data[i];
+   }
+#endif
+
+   hypre_SyncComputeStream(hypre_handle());
```

Comment:
@liruipeng I put this synchronization here since I saw it's being used in the other functions as well. Thinking twice, it would be good to remove it and deal with synchronization outside of the function so it becomes more generic. What are your thoughts?

Reply:
Which place outside the functions do you want to put the sync?

Reply:
I would put in the function that calls `hypre_ParVectorAxpyz` depending on whether a synchronization is necessary or not. The idea would be to make these basic linear algebra functions more flexible by giving the user the ability to decide whether a synchronization must be called or not. To make it safe to external users we could impose the synchronization in the public `HYPRE_` functions (when they exist). The downside of this approach would be that ,inside hypre, we should not take it for granted that device synchronization is called anymore in these BLAS-like functions. Thus, we would need to explicitly write the synchronization calls when necessary... In summary, I see advantages and disadvantages and I believe we can discuss that :) @rfalgout